numpy/lib/arraysetops.py

```diff
@@ -223,10 +223,11 @@ def unique(ar, return_index=False, return_inverse=False,
         ret = _unique1d(ar, return_index, return_inverse, return_counts)
         return _unpack_tuple(ret)
 
-    if not (-ar.ndim <= axis < ar.ndim):
-        raise ValueError('Invalid axis kwarg specified for unique')
+    try:
+        ar = np.swapaxes(ar, axis, 0)
+    except np.AxisError:
+        raise np.AxisError(axis, ar.ndim)
```

Comment:
This makes the error message not include `axis1` or `axis2`

Reply:
If you needed to explain it in the PR, write this as a comment in the code instead :)